### PR TITLE
Add "WaitForReady" method

### PR DESCRIPTION
**Pull request by samsondav**
This couples with the non-blocking start so that subsequent calls can
wait for the dial to finish.

### CODE REVIEW

**Review by jkongie — APPROVED** [review]

File: dialoptions.go [hunk 1]
```diff
@@ -74,8 +76,15 @@ func WithWriteTimeout(d time.Duration) DialOption {
 	})
 }
 
+func WithLogger(lggr logger.Logger) DialOption {
```
Comment (samsondav) [inline dialoptions.go:79]: Actually might be preferable to just make it required.

File: client.go [hunk 2]
```diff
@@ -454,7 +479,7 @@ func (ac *addrConn) resetTransport() {
 			ac.mu.Unlock()
 
 			// Reconnection backoff time
-			log.Println("[wsrpc] attempting reconnection in", backoffFor)
+			ac.dopts.logger.Infof("attempting reconnection in %s", backoffFor)
```
Comment (jkongie) [inline client.go:482]: Is it ok to use `%s` here? `backoffFor` is a `time.Duration`

File: dialoptions.go (hunk 1 above)
Comment (jkongie) [reply dialoptions.go:79]: I'd prefer to not make backwards incompatible changes here

**Review by jkongie — COMMENTED** [review]
Looks good. Just one question on a log statement but otherwise looks good

File: client.go (hunk 2 above)
Comment (samsondav) [reply client.go:482]: Yeah it will call `.String()` and print a human readable duration

Comment (samsondav) [reply client.go:482]: e.g. https://go.dev/play/p/P9uijy4cDg6

**Review by jkongie — APPROVED** [review]

**Pull request merged**